src/components/dialogs/network-modifications/generator/modification/generator-modification-form.jsx

```diff
@@ -51,6 +56,47 @@ const GeneratorModificationForm = ({
           })
         : undefined;
 
+    const watchVoltageRegulation = useWatch({
+        name: VOLTAGE_REGULATION,
+    });
+
+    const previousRegulation = () => {
+        if (generatorToModify?.voltageRegulatorOn) {
+            return intl.formatMessage({ id: 'On' });
+        }
+        if (generatorToModify?.voltageRegulatorOn === false) {
+            return intl.formatMessage({ id: 'Off' });
+        }
+        return null;
+    };
+
+    const voltageRegulationField = (
```

Comment:
Why do we have 2 components ? We could have a VoltageRegulationForm component with the switch and the fields nope ?
Or sometimes we have only the switch or the fields ?